test/governance/candidacy.users.test.ts

```diff
@@ -0,0 +1,233 @@
+import { Keyring } from "@polkadot/api";
+import { Bank } from "../../utils/Framework/Supply/Bank";
+import { Node } from "../../utils/Framework/Node/Node";
+import { SudoUser } from "../../utils/Framework/User/SudoUser";
+import { UserFactory, Users } from "../../utils/Framework/User/UserFactory";
+import { cryptoWaitReady } from "@polkadot/util-crypto";
+import { getEnvironmentRequiredVars, waitForNBlocks } from "../../utils/utils";
+
+// Global variables
+const { chainUri: environmentUri } = getEnvironmentRequiredVars();
+jest.spyOn(console, "log").mockImplementation(jest.fn());
+jest.setTimeout(15000000);
+
+// Chain variables
+let bootnode: Node;
+let keyring: Keyring;
+let sudo: SudoUser;
+let termDuration: number; // Set as a const in mangata-node
+const termBlockDelta = 3; // A small block delta at term boundaries
+
+// Test variables
+let candidate: GovernanceUser;
+let voter: GovernanceUser;
+let bank: Bank;
+const stakeAmount = new BN(Math.pow(10, 16).toString());
+const accountFundingAmount = new BN(Math.pow(10, 17).toString());
+
+// Cross-test state
+const state: {
+  accounts: [GovernanceUser | null];
+} = { accounts: [null] };
+
+beforeAll(async () => {
+  await cryptoWaitReady(); // Wait for Polkadots WASM backend
+
+  // Instantiate nodes
+  bootnode = new Node(environmentUri);
+  await bootnode.connect();
+  await bootnode.subscribeToHead();
+
+  // Set chain variables
+  termDuration = (
+    await bootnode.api!.derive.elections.info()
+  ).termDuration.toNumber();
+  keyring = new Keyring({ type: "sr25519" });
+  sudo = UserFactory.createUser(Users.SudoUser, keyring, bootnode) as SudoUser;
+});
+
+beforeEach(async () => {
+  // Don't run tests at the end of a term
+  if (termDuration % bootnode.lastBlock!) {
+    waitForNextTerm();
+  }
+
+  // Initialize actors
+  candidate = UserFactory.createUser(
+    Users.GovernanceUser,
+    keyring,
+    bootnode
+  ) as GovernanceUser;
+  voter = UserFactory.createUser(
+    Users.GovernanceUser,
+    keyring,
+    bootnode
+  ) as GovernanceUser;
+  bank = new Bank(sudo);
+
+  // Record new test state
+  state.accounts.push(voter);
+
+  // Fund accounts
+  await voter.addMGATokens(bank.sudoUser, accountFundingAmount);
+  await candidate.addMGATokens(bank.sudoUser, stakeAmount);
+
+  // Subscribe to events
+  candidate.node.subscribeToUserBalanceChanges(candidate);
+  voter.node.subscribeToUserBalanceChanges(voter);
+});
+
+afterAll(async () => {
+  // Tests in files are unlikely to impact each other
+  // but not clearing up over multiple test files will break
+  await cleanState();
+
+  // Stops duplicated bootnode subscriptions creating duplicate logs
+  await bootnode.stop();
+});
+
+describe("Governance -> Candidacy -> Users", () => {
+  test("User can run for candidacy", async () => {
+    const candidatesAddress = candidate.keyRingPair.address;
+
+    await candidate.runForCouncil();
+    await waitForNBlocks(2);
+
+    const termZero = getLastBlocksElectionData();
```

Comment:
All the assertions are done for Block events. Can you please incorporate also that:
User must / must not be in the candidate/members list?
There are some chain-state data retrieval requests. 

`elections.candidates: Vec<AccountId>`
`members(): Vec<(AccountId,BalanceOf)>`

Reply:
This data comes from state queries 

```
        await this.api!.queryMulti(
          [
            this.api!.query.elections.candidates,
            this.api!.query.elections.members,
          ],
          ([candidates, members]) => {
            this.electionEvents.set(this.lastBlock!, {
              candidates: candidates.toJSON(),
              members: members.toJSON(),
            });
            ```